components/scream/scripts/test_all_scream.py

```diff
@@ -73,12 +98,19 @@ def __init__(self, cxx, kokkos, submit, parallel, fast_fail, baseline_ref, basel
                 self._proc_count = 1
 
         if self._keep_tree:
-            if self._baseline_dir=="NONE":
+            expect(not is_repo_clean(), "Makes no sense to use --keep-tree when repo is clean")
```

Comment:
Sure, it makes no sense, but should we really error out? Perhaps just print a warning?

Reply:
I think we should error out; the user is likely confused and is not testing what they think they are testing.

Reply:
Fair enough.

Reply:
I'm not a python guru, so this may be naive: the expect is not indented inside the if. Is that a problem?

Reply:
The expect is part of the block that gets executed if self._keep_tree is true.

Reply:
Oh, sorry. I didn't realized the line above is red, hence removed.